enumerables.rb

```diff
@@ -0,0 +1,153 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < to_a.length
+      yield to_a[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < to_a.length
+      yield to_a[i], i
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    result = []
+    to_a.my_each { |item| result << item if yield(item) }
+    result
+  end
+
+  def my_all?
+    to_a.my_each do |i|
+      return false if yield(i) == false
+      else
+        return true
+      end
+    end
+  end
+
+  def my_any?
+    return to_enum(:my_any?) unless block_given?
+
+    i = 0
+    ans = false
+    to_a.length.times do
+      ans = yield to_a[i]
+
+      break if ans == true
+
+      i += 1
+    end
+    ans
+  end
+
+  def my_none?
+    return to_enum(:my_none?) unless block_given?
+
+    i = 0
+    ans = false
+    to_a.length.times do
+      ans = yield to_a[i]
+
+      break if ans == true
+
+      i += 1
+    end
+    !ans
+  end
+
+  def my_count(number = nil)
+    count = 0
+    if number.nil?
+      count = to_a.length
+    else
+
+      my_each { |i| count += 1 if number == i }
+
+      print count
+    end
+  end
+
+  def my_map(proc = nil)
+    result = []
+
+    to_a.my_each { |item| result << item if proc.call(item) }
+    result
+  end
+
+  def my_inject(number = nil, symbol = nil)
+    # Symbol only
+    if !number.nil? && symbol.nil? && number.is_a?(String) || number.is_a?(Symbol)
+      array = to_a
+      product = array[0]
+      (array.length - 1).times do |i|
+        product = array[i + 1].send(number[0], product)
+      end
+      product
+
+    # block only
+    elsif block_given?
+      total = 1
+      my_each do |i|
+        total = yield(total, i)
+      end
+      total
+    elsif !number.nil? && symbol.nil?
+      array = to_a
+      product = number
+      array.length(-1).times do |i|
+        product = array[i + 1].send(number[0], product)
+      end
+      product
+    end
+  end
+end
```

Comment:
***my_inject***

I will be using these variables for the next suggestions:

```
array = [2, 5, 9, 7, 2, 2]
words = %w[dog door rod blade]
search = proc { |memo, word| memo.length > word.length ? memo : word } 
Range.new(5, 50)
``` 

- [ ] Raises a "LocalJumpError" when no block or argument is given

```
      array.my_inject

        expected LocalJumpError but nothing was raised
```

- [ ] Searches for the longest word in an array of strings

```
      words.my_inject(&search)
      
      NoMethodError:
        undefined method `length' for 1:Integer
      # ./enumerables.rb:102:in `block in my_inject'
      # ./enumerables.rb:7:in `my_each'
      # ./enumerables.rb:101:in `my_inject'
```

- [ ] When a symbol is specified with an initail value combines each element of the collection by applying the symbol as a named method:

```
        range.my_inject(2, :*)
      
        Expected: 2534507766809448170301050680505397403698136797413376000000000000
        Got: nil
```